### DISCUSSION TITLE
Is there a way to paginate get_detect_summaries?

### DISCUSSION THREAD

**Discussion by brucelourenco** [CrowdStrike/falconpy#1111] [Q&A] [answered]
Hello community, I hope all of you are doing well.
I was searching for a solution, besides Event Stream, to get more than 1k of detections summaries from (`GetDetectSummaries`) and I couldn't find out an answer yet.
In the docus I could see the offset/limit parameters for (`QueryDetects`) but it does't exists for (`GetDetectSummaries`).
Could you guys give me some advice to achieve this?
Thank you!

**Comment by jshcodes** [comment]
Hi @brucelourenco - 

[`GetDetectSummaries`](https://www.falconpy.io/Service-Collections/Detects.html#getdetectsummaries) is not a composite endpoint, so it does not perform filtering or support parameters like `offset` and `limit`.

[`GetDetectSummaries`](https://www.falconpy.io/Service-Collections/Detects.html#getdetectsummaries) supports up to __1,000__ IDs per request. If the final list (after paginating using [`QueryDetects`](https://falconpy.io/Service-Collections/Detects.html#querydetects)) is going to consistently exceed this, you might consider using something like [`concurrent.futures`](https://docs.python.org/3/library/concurrent.futures.html) to speed up processing.

**Reply by jshcodes** [reply, marked as answer]
Here's a quick example. 😄 

```python
import logging
from concurrent.futures import ThreadPoolExecutor
from falconpy import Detects


def get_detects_details(detect_list: list):
    return falcon.get_detect_summaries(ids=detect_list)["body"]["resources"]

logging.basicConfig(level=logging.DEBUG)

falcon = Detects(debug=True)  # Pass client_id and client_secret if they are not in your environment

offset = 0
limit = 9999
detection_ids = []

while True:
    result = falcon.query_detects(offset=offset, limit=limit)["body"]
    total = result["meta"]["pagination"]["total"]
    offset = result["meta"]["pagination"]["offset"]
    detection_ids.extend(result["resources"])
    if len(detection_ids) >= total:
        break

print(f"{len(detection_ids)} total detections found")

batches = [detection_ids[i:i+1000] for i in range(0, len(detection_ids), 1000)]
details = []
with ThreadPoolExecutor() as executor:
    futures = {
        executor.submit(get_detects_details, batch) for batch in batches
    }
    for fut in futures:
        details.extend(fut.result())


print(f"{len(details)} total detections enriched")

```

**Reply by brucelourenco** [reply]
Hello @jshcodes ! Excellent!
Thanks a lot for your tips
